test/test_baseline_trainers.py

```diff
@@ -89,23 +88,77 @@ def test_trainers(test_cfg_path, mode, gpu2gpu, observation_transforms):
         if not habitat_sim.cuda_enabled:
             pytest.skip("GPU-GPU requires CUDA")
 
-    run_exp(
-        test_cfg_path,
-        mode,
-        [
-            "TASK_CONFIG.SIMULATOR.HABITAT_SIM_V0.GPU_GPU",
-            str(gpu2gpu),
-            "RL.POLICY.OBS_TRANSFORMS.ENABLED_TRANSFORMS",
-            str(tuple(observation_transforms)),
-        ],
-    )
+    try:
+        run_exp(
+            test_cfg_path,
+            mode,
+            [
+                "TASK_CONFIG.SIMULATOR.HABITAT_SIM_V0.GPU_GPU",
+                str(gpu2gpu),
+                "RL.POLICY.OBS_TRANSFORMS.ENABLED_TRANSFORMS",
+                str(tuple(observation_transforms)),
+            ],
+        )
+    finally:
+        # Needed to destroy the trainer
+        gc.collect()
 
-    # Needed to destroy the trainer
-    gc.collect()
+        # Deinit processes group
+        if torch.distributed.is_initialized():
+            torch.distributed.destroy_process_group()
```

Comment:
Probably out of scope here, but should this try/finally be part of `run_exp` or `execute_exp`  ?